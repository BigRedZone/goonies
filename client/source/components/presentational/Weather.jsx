import React from 'react';
import axios from 'axios';

class Weather extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentTemp: '',
      currentClouds: '',
      currentHumidity: '',
      currentWind: '',
      currentLocation: '',
      // fiveDayForecast: [],
    };
  }

  componentDidMount() {
    this.getCurrentWeatherData();
    this.getFiveDayForcast();
  }

  getCurrentWeatherData() {
    axios.get('/api/weathercurrent')
      .then((response) => {
        console.log('responsed Current', response.data);
        this.setState({
          currentTemp: response.data.main.temp,
          currentClouds: response.data.clouds.all,
          currentHumidity: response.data.main.humidity,
          currentWind: response.data.wind.speed,
          currentLocation: response.data.name,
          // fiveDayForecast: [],
        });
      })
      .catch((error) => {
        console.log(error);
      });
  }
  // HAS 8 FORCATS PER DAY EVERY 3 HOURS

  getFiveDayForcast() {
    axios.get('/api/weatherfive')
      .then((response) => {
        console.log('responded with 5 day', response.data);
        this.setState({
          // fiveDayForecast: response.data,
        });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    const {
      currentTemp,
      currentClouds,
      currentHumidity,
      currentWind,
      currentLocation,
    } = this.state;
    return (
      <div>
        <div>
          <h2>currentWeather</h2>
          <h4>{currentLocation}</h4>
          <p>
            Temp:
            {' '}
            {currentTemp}
            {' '}
            FÂ°
          </p>
          <p>
            Clouds:
            {' '}
            {currentClouds}
            %
          </p>
          <p>
            Humidity:
            {' '}
            {currentHumidity}
            %
          </p>
          <p>
            Wind:
            {' '}
            {currentWind}
            {' '}
            mph
          </p>
        </div>

        <h2>5-Day Forcast</h2>
      </div>
    );
  }
}

export default Weather;
