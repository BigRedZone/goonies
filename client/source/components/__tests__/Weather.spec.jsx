import React from 'react';
import renderer from 'react-test-renderer';
import Weather from '../presentational/Weather';

test('App Renders correctly', () => {
  const component = renderer.create(<Weather />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
