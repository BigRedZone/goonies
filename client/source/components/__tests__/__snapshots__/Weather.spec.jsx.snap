// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders correctly 1`] = `
<div>
  <div>
    <h2>
      Current Weather
    </h2>
    <h4>
      
    </h4>
    <h3>
      
    </h3>
    <p>
      Temp:
       
      
       
      F°
    </p>
    <p>
      Clouds:
       
      
      %
    </p>
    <p>
      Humidity:
       
      
      %
    </p>
    <p>
      Wind:
       
      
       
      mph
    </p>
  </div>
  <div>
    <div>
      <h2>
        3 Day Forecast
      </h2>
      <h3>
        Tomorrow
      </h3>
      <h4>
        
      </h4>
      <p>
        High Temp:
         
        
         
        F°
      </p>
      <p>
        Low Temp:
         
        
         
        F°
      </p>
      <h3>
        Day 2
      </h3>
      <h4>
        
      </h4>
      <p>
        High Temp:
         
        
         
        F°
      </p>
      <p>
        Low Temp:
         
        
         
        F°
      </p>
      <h3>
        Day 3
      </h3>
      <h4>
        
      </h4>
      <p>
        High Temp:
         
        
         
        F°
      </p>
      <p>
        Low Temp:
         
        
         
        F°
      </p>
    </div>
  </div>
</div>
`;
